capstone/src/main/java/com/google/step/clients/ScoreClient.java

```diff
@@ -0,0 +1,160 @@
+ */
+public class ScoreClient {
+
+    private final float minimum = 10.0f;
+    private final MetricsClient metricsClient = new MetricsClient();
+
+    /**
+     * Only method to be called and calculates the scores of each restaurant.
+     * @return an unmodifiable list of restaurantScore class.
+     */
+    public List<RestaurantScore> calculateScores() {
+        Map<Long, Double> scoreMap = new HashMap<>();
+        List<Long> restaurantIds = new ArrayList<>();
+
+        List<RestaurantPageViews> allPageViews = metricsClient.getAllPageViews("restaurantKey");
+
+        double systemAverage = getSystemAverage(allPageViews);
+
+        // Calculate score
+        for (RestaurantPageViews restaurant: allPageViews) {
+            double restaurantAverage = calculateRestaurantAveragePageViews(restaurant);
+
+            // Since the data is sorted we can just get the latest pageView for that restaurant.
+            int lastPageViewIndex = restaurant.getPageViews().size() - 1;
+            int latestPageView = restaurant.getPageViews().get(lastPageViewIndex).getCount();
+
+            double score = calculateRawScore(latestPageView, restaurantAverage, systemAverage);
+            restaurantIds.add(Long.parseLong(restaurant.getName()));
+            scoreMap.put(Long.parseLong(restaurant.getName()), score);
+        }
+
+        standardizeScores(scoreMap, restaurantIds);
+        normalizeScores(scoreMap, restaurantIds);
+
+        // Build the scores return list.
+        List<RestaurantScore> scores = new ArrayList<>();
+        for (Long restaurantId: restaurantIds) {
+            scores.add(new RestaurantScore(restaurantId, scoreMap.get(restaurantId)));
+        }
+
+        return Collections.unmodifiableList(scores);
+    }
+
+    /**
+     * Gets the average pageViews for the entire system.
+     * @param allPageViews list of restaurantPageViews holding all the pageViews.
+     * @return double representing the average pageViews for the system.
+     */
+    private double getSystemAverage(List<RestaurantPageViews> allPageViews) {
+        double sum = 0;
+        double count = 0;
+
+        for (RestaurantPageViews restaurant : allPageViews) {
+            for (int i = 0; i < restaurant.getPageViews().size(); i++) {
+                sum += restaurant.getPageViews().get(i).getCount();
+                count++;
+            }
+        }
+
+        return sum / count;
+    }
+
+    /**
+     * Calculate the average page views for a specific restaurant.
+     * @param restaurant has all the pageViews for a specific restaurant.
+     * @return double representing the average pageViews for that restaurant.
+     */
+    private double calculateRestaurantAveragePageViews(RestaurantPageViews restaurant) {
+        double sum = 0;
+        double count = 0;
+
+        for (int i = 0; i < restaurant.getPageViews().size(); i++) {
+            sum += restaurant.getPageViews().get(i).getCount();
+            count++;
```

Comment:
Nit: you could probably just store the value of `restaurant.getPageViews().size()` in a variable and use that as `count` instead.

Reply:
Done